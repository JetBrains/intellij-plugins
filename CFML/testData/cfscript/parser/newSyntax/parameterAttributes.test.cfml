component rest="true" restpath="widget"{

  remote string function getWidget(string widgetid restargsource="path") httpmethod="GET" restpath="{widgetid}" {

  return arguments.widgetid;
}

}