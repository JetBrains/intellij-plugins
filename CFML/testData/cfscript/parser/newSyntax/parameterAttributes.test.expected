CfmlFile:parameterAttributes.cfml
  COMPONENT_DEFINITION
    PsiElement(COMPONENT_KEYWORD)('component')
    Attribute
      PsiElement(IDENTIFIER)('rest')
      PsiElement(ASSIGN)('=')
      PsiElement(DOUBLE_QUOTE)('"')
      AttributeValue
        PsiElement(STRING_TEXT)('true')
      PsiElement(DOUBLE_QUOTE_CLOSER)('"')
    Attribute
      PsiElement(IDENTIFIER)('restpath')
      PsiElement(ASSIGN)('=')
      PsiElement(DOUBLE_QUOTE)('"')
      AttributeValue
        PsiElement(STRING_TEXT)('widget')
      PsiElement(DOUBLE_QUOTE_CLOSER)('"')
    PsiElement(L_CURLYBRACKET)('{')
    FunctionDefinition
      PsiElement(REMOTE_KEYWORD)('remote')
      TYPE
        PsiElement(STRING_TYPE)('string')
      PsiElement(FUNCTION_KEYWORD)('function')
      PsiElement(IDENTIFIER)('getWidget')
      ParametersList
        PsiElement(L_BRACKET)('(')
        FUNCTION_ARGUMENT
          TYPE
            PsiElement(STRING_TYPE)('string')
          PsiElement(IDENTIFIER)('widgetid')
          Attribute
            PsiElement(IDENTIFIER)('restargsource')
            PsiElement(ASSIGN)('=')
            PsiElement(DOUBLE_QUOTE)('"')
            AttributeValue
              PsiElement(STRING_TEXT)('path')
            PsiElement(DOUBLE_QUOTE_CLOSER)('"')
        PsiElement(R_BRACKET)(')')
      Attribute
        PsiElement(IDENTIFIER)('httpmethod')
        PsiElement(ASSIGN)('=')
        PsiElement(DOUBLE_QUOTE)('"')
        AttributeValue
          PsiElement(STRING_TEXT)('GET')
        PsiElement(DOUBLE_QUOTE_CLOSER)('"')
      Attribute
        PsiElement(IDENTIFIER)('restpath')
        PsiElement(ASSIGN)('=')
        PsiElement(DOUBLE_QUOTE)('"')
        AttributeValue
          PsiElement(STRING_TEXT)('{widgetid}')
        PsiElement(DOUBLE_QUOTE_CLOSER)('"')
      FUNCTIONBODY
        BLOCK_OF_STATEMENTS
          PsiElement(L_CURLYBRACKET)('{')
          PsiElement(RETURN_KEYWORD)('return')
          ReferenceExpression
            PsiElement(SCOPE_KEYWORD)('arguments')
            PsiElement(POINT)('.')
            PsiElement(IDENTIFIER)('widgetid')
          PsiElement(SEMICOLON)(';')
          PsiElement(R_CURLYBRACKET)('}')
    PsiElement(R_CURLYBRACKET)('}')